server/src/models/itemModel.js

```diff
@@ -6,24 +6,25 @@ export function initDb() {
   db = initDBConnection();
 }
 
-export const itemsByCategory = async (id) => {
+export const itemsByCategory = async () => {
   const { rows: items } = await db.query(
-    `SELECT item.id, item.category_id, item.item_name, item.price, category.category_name, category.description
-    FROM 
-    item INNER JOIN 
-    category on item.category_id = category.id 
-    WHERE category_id = ${id} AND item.active = TRUE
-    ORDER BY id ASC;`
+    `SELECT
+    C.ID,
+    C.CATEGORY_NAME as categoryName ,
+    C.DESCRIPTION,
+   (
+        SELECT json_agg(json_build_object(
+            'id', ID,
+            'itemName', ITEM_NAME,
+            'price', PRICE
+        ))
+        FROM ITEM
+        WHERE
+            CATEGORY_ID = C.ID AND ACTIVE IS TRUE
+    ) AS  ITEM    
```

Comment:
This would be a clearer name because it's an array of items.

```suggestion
    ) AS  ITEMS
```